src/habitat.cr

```diff
@@ -37,6 +37,64 @@ class Habitat
     end
   end
 
+  # Embed settings in a Class or Module.
+  #
+  # A class or module can call `Habitat.create` with a block of `setting` calls
+  # that will declare the types (and optionally default values) of our settings.
+  #
+  # ```
+  # class MyServer
+  #   Habitat.create do
+  #     setting port : Int32
+  #     setting debug_errors : Bool = true
+  #   end
+  # end
+  # ```
+  #
+  # `create` adds class and instance `settings` methods to the embedding
+  # class/module, which we'll use to set and get the values of our settings.
+  #
+  # ```
+  # MyServer.settings.port = 80
+  # MyServer.settings.port # 80
+  # ```
```

Comment:
It would be worth noting that you can also get at settings at an instance level:

```
class MyServer
  def start
    settings.port # => 80
  end
end
```